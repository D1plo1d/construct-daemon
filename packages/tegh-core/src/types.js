/* config */
export Config, { MockConfig } from './config/types/Config'
export * as AxisTypeEnum from './config/types/AxisTypeEnum'

/* jobQueue */
export * as JobHistoryTypeEnum from './jobQueue/types/JobHistoryTypeEnum'

export JobHistoryEvent from './jobQueue/types/JobHistoryEvent'
export JobQueueGraphQL from './jobQueue/types/JobQueue.graphql.js'
export Job, { MockJob } from './jobQueue/types/Job'
export JobGraphQL from './jobQueue/types/Job.graphql.js'
export JobFile, { MockJobFile } from './jobQueue/types/JobFile'

/* printer */
export PrinterGraphQL from './printer/types/Printer.graphql.js'

/* spool */
export * as PriorityEnum from './spool/types/PriorityEnum'
export * as TaskStatusEnum from './spool/types/TaskStatusEnum'
export * as ComponentTypeEnum from './config/types/ComponentTypeEnum'

export Task, { MockTask } from './spool/types/Task'
export TaskGraphQL from './spool/types/Task.graphql.js'
