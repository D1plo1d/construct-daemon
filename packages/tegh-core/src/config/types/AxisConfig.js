import { Record } from 'immutable'
import uuid from 'uuid/v4'

export const AxisConfigFactory = Record({
  id: null,
  name: null,
  feedrate: null,
})

const AxisConfig = props => (
  AxisConfigFactory({
    id: props.id || uuid(),
    ...props(),
  })
)

export default AxisConfig
