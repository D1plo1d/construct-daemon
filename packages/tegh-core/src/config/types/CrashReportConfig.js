import { Record, Map } from 'immutable'
import uuid from 'uuid'

export const CrashReportConfigFactory = Record({
  id: null,
  modelVersion: 0,
  directory: '/var/log/tegh',
  uploadCrashReportsToDevs: true,
  extendedConfig: Map(),
})

const CrashReportConfig = ({
  id = uuid.v4(),
  modelVersion = 0,
  ...props
} = {}) => (
  CrashReportConfigFactory({
    ...props,
    id,
    modelVersion,
    extendedConfig: Map(props.extendedConfig),
  })
)

export default CrashReportConfig
