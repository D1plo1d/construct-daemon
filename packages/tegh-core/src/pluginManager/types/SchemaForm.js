import { Record, Map } from 'immutable'

const SchemaForm = Record({
  id: null,
  schema: null,
  form: null,
  configPaths: Map({}),
})

export const NullSchemaForm = SchemaForm({
  id: 'NULL',
  schema: {},
  form: [],
})

export default SchemaForm
