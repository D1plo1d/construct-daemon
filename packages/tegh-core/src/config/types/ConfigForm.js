import { Record, Map } from 'immutable'
import t from 'tcomb-validation'

import LogConfig, { LogConfigStruct } from './LogConfig'
import CrashReportConfig, { CrashReportConfigStruct } from './CrashReportConfig'
import PrintFromLocalPathConfig, { PrintFromLocalPathConfigStruct } from './PrintFromLocalPathConfig'
import PluginConfig, { PluginConfigStruct } from './PluginConfig'
import MachineConfig, { MachineConfigStruct } from './MachineConfig'
import MaterialConfig, { MaterialConfigStruct } from './MaterialConfig'

export const configFormStructFields = {
  id: t.String,
  name: t.String,
  macros: t.dict(t.String, t.list(t.String)),
  log: LogConfigStruct,
  crashReports: CrashReportConfigStruct,
  printFromLocalPath: PrintFromLocalPathConfigStruct,
  materials: t.dict(t.String, MaterialConfigStruct),
  machine: MachineConfigStruct,
  plugins: t.dict(t.String, PluginConfigStruct),
}

export const ConfigFormStruct = t.struct(configFormStructFields)

export const ConfigFormRecordFactory = Record(
  Map(ConfigFormStruct.meta.props).map(() => null),
)

const mapOfRecords = (entries, recordFactory) => (
  Map(entries).map(props => recordFactory(props))
)

const ConfigForm = props => ConfigFormRecordFactory({
  ...props,
  log: LogConfig(props.log),
  crashReports: CrashReportConfig(props.crashReports),
  printFromLocalPath: PrintFromLocalPathConfig(props.printFromLocalPath),
  plugins: mapOfRecords(props.plugins, PluginConfig),
  machine: MachineConfig(props.machine),
  materials: mapOfRecords(props.materials, MaterialConfig),
})

export default ConfigForm
