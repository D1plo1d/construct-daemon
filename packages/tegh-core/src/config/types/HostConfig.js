import { Record, List, Map } from 'immutable'
import uuid from 'uuid/v4'

import CrashReportConfig from './CrashReportConfig'
import MaterialConfig from './MaterialConfig'
import LogConfig from './LogConfig'

export const HostConfigRecordFactory = Record({
  id: null,
  modelVersion: 0,
  name: null,
  crashReports: CrashReportConfig(),
  materials: List(),
  extendedConfig: Map(),
  log: LogConfig(),
})

const HostConfig = ({
  id = uuid(),
  modelVersion = 0,
  crashReports = {},
  materials = [],
  log = {},
  ...props
} = {}) => (
  HostConfigRecordFactory({
    ...props,
    id,
    modelVersion,
    crashReports: CrashReportConfig(crashReports),
    materials: materials.map(MaterialConfig),
    log: LogConfig(log),
    extendedConfig: Map(props.extendedConfig),
  })
)

export default HostConfig
