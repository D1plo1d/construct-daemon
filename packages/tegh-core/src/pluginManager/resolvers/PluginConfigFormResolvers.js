import { NullSchemaForm } from '../types/SchemaForm'

const PluginConfigFormResolvers = {
  PluginConfigForm: {
    schemaForm: (source, args, { store }) => {
      const state = store.getState()
      return state.schemaForms.getIn(
        ['plugins', source.package],
        NullSchemaForm,
      )
    },
  },
}

export default PluginConfigFormResolvers
