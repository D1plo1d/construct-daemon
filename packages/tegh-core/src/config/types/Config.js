import { Record, List } from 'immutable'
import uuid from 'uuid/v4'

import HostConfig from './HostConfig'
import PrinterConfig from './PrinterConfig'
import MaterialConfig from './MaterialConfig'
import developmentConfig from '../../../../../development.config'

export const ConfigRecordFactory = Record({
  id: null,
  modelVersion: 0,
  host: null,
  printer: null,
  materials: List(),
})

const Config = ({
  id = uuid(),
  modelVersion = 0,
  host = {},
  printer = {},
  materials = [],
  ...props
} = {}) => (
  ConfigRecordFactory({
    id,
    modelVersion,
    host: HostConfig(host),
    printer: PrinterConfig(printer),
    materials: List(materials).map(material => MaterialConfig(material)),
    ...props,
  })
)

export const MockConfig = ({
  host = {},
  printer = {},
  ...props
} = {}) => (
  Config({
    ...developmentConfig,
    host: {
      ...developmentConfig.host,
      ...host,
    },
    printer: {
      ...developmentConfig.printer,
      ...printer,
    },
    ...props,
  })
)

export default Config
