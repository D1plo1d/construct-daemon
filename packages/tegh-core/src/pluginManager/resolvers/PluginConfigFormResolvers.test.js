import { Record, Map } from 'immutable'
import typeDefs from 'tegh-schema'

import deterministicTestSetup from '../../util/testing/deterministicTestSetup'
import snapshotTestResolvers from '../../util/testing/snapshotTestResolvers'

import { MockConfig } from '../../config/types/Config'
import SchemaForm from '../types/SchemaForm'

import PluginConfigFormResolvers from './PluginConfigFormResolvers'

describe('PluginConfigFormResolvers', () => {
  it('presents a consistent API', () => {
    deterministicTestSetup()

    const state = Record({
      schemaForms: Record({
        plugins: Map({
          'tegh-core': SchemaForm({
            id: 'tegh-core',
          }),
        }),
      })(),
    })()

    snapshotTestResolvers({
      typeDefs,
      typeName: 'PluginConfigForm',
      resolvers: PluginConfigFormResolvers,
      rootValue: MockConfig().printer.plugins.find(p => (
        p.package === 'tegh-core'
      )),
      contextValue: {
        store: {
          getState: () => state,
        },
      },
    })
  })
})
