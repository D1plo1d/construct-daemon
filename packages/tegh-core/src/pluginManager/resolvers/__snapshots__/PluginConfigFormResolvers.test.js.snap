// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginConfigFormResolvers presents a consistent API 1`] = `
Object {
  "_testPluginConfigForm": Object {
    "id": "asgbvas23",
    "model": Immutable.Map {
      "modelID": "lulzbot/lulzbot-mini-2",
      "name": "Left Home Lulzbot",
    },
    "modelVersion": 1,
    "package": "tegh-core",
    "schemaForm": Object {
      "id": "tegh-core",
    },
  },
}
`;
